SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialTraceEventBuilder.h

```diff
@@ -0,0 +1,237 @@
+	FSpatialTraceEventBuilder AddComponentId(FString Key, const Worker_ComponentId ComponentId);
+	FSpatialTraceEventBuilder AddFieldId(FString Key, const uint32 FieldId);
+	FSpatialTraceEventBuilder AddNewWorkerId(FString Key, const uint32 NewWorkerId);
+	FSpatialTraceEventBuilder AddCommand(FString Key, const FString& Command);
+	FSpatialTraceEventBuilder AddRequestID(FString Key, const int64 RequestID);
+	FSpatialTraceEventBuilder AddAuthority(FString Key, const Worker_Authority Role);
+	FSpatialTraceEventBuilder AddKeyValue(FString Key, FString Value);
+	FSpatialTraceEvent GetEvent() &&;
+
+	// --- Static Functions ---
+
+	static FSpatialTraceEvent ProcessRPC(const UObject* Object, UFunction* Function)
+	{
+		return FSpatialTraceEventBuilder(GDK_EVENT_NAMESPACE "process_rpc")
+			.AddObject(TEXT("Object"), Object)
+			.AddFunction(TEXT("Function"), Function)
+			.GetEvent();
+	}
+
+	static FSpatialTraceEvent SendRPC(const UObject* Object, UFunction* Function)
+	{
+		return FSpatialTraceEventBuilder(GDK_EVENT_NAMESPACE "send_rpc")
+			.AddObject(TEXT("Object"), Object)
+			.AddFunction(TEXT("Function"), Function)
+			.GetEvent();
+	}
+
+	static FSpatialTraceEvent QueueRPC() { return FSpatialTraceEventBuilder("queue_rpc").GetEvent(); }
+
+	static FSpatialTraceEvent RetryRPC() { return FSpatialTraceEventBuilder("retry_rpc").GetEvent(); }
+
+	static FSpatialTraceEvent PropertyUpdate(const UObject* Object, const Worker_EntityId EntityId, const Worker_ComponentId ComponentId,
+											 const FString& PropertyName)
+	{
+		return FSpatialTraceEventBuilder(GDK_EVENT_NAMESPACE "property_update")
+			.AddObject(TEXT("Object"), Object)
+			.AddEntityId(TEXT("EntityId"), EntityId)
+			.AddComponentId(TEXT("ComponentId"), ComponentId)
+			.AddKeyValue("PropertyName", PropertyName)
+			.GetEvent();
+	}
+
+	static FSpatialTraceEvent MergeComponent(const Worker_EntityId EntityId, const Worker_ComponentId ComponentId)
+	{
+		return FSpatialTraceEventBuilder(GDK_EVENT_NAMESPACE "merge_component")
+			.AddEntityId(TEXT("EntityId"), EntityId)
+			.AddComponentId(TEXT("ComponentId"), ComponentId)
+			.GetEvent();
+	}
+
+	static FSpatialTraceEvent SendCommandRequest(const FString& Command, const int64 RequestID)
+	{
+		return FSpatialTraceEventBuilder(GDK_EVENT_NAMESPACE "send_command_request")
+			.AddCommand(TEXT("Command"), Command)
+			.AddRequestID(TEXT("RequestID"), RequestID)
+			.GetEvent();
+	}
+
+	static FSpatialTraceEvent ReceiveCommandRequest(const FString& Command, const int64 RequestID)
+	{
+		return FSpatialTraceEventBuilder(GDK_EVENT_NAMESPACE "receive_command_request")
+			.AddCommand(TEXT("Command"), Command)
+			.AddRequestID(TEXT("RequestID"), RequestID)
+			.GetEvent();
+	}
+
+	static FSpatialTraceEvent ReceiveCommandRequest(const FString& Command, const UObject* Actor, const UObject* TargetObject,
+													const UFunction* Function, const int32 TraceId, const int64 RequestID)
```

Comment:
super nit - we tend to use "Id" instead of "ID". Even thought we use UpperPascal case everwhere.